janus_server/src/config.rs

```diff
@@ -0,0 +1,110 @@
+//! Configuration for various Janus actors
+
+use crate::message::{Duration, HpkeConfig, Role, TaskId};
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use url::Url;
+
+/// Configuration for a Janus server using a database.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DbConfig {
+    /// URL at which to connect to the database.
+    pub url: Url,
+    // TODO: add option for connecting to database over TLS, if necessary
+}
+
+/// Non-secret configuration options for a Janus aggregator, deserialized from
+/// YAML.
+///
+/// # Examples
+///
+/// ```
+/// use janus_server::config::AggregatorConfig;
+///
+/// let yaml_config = r#"
+/// ---
+/// listen_address: "0.0.0.0:8080"
+/// database:
+///   url: "postgres://postgres:postgres@localhost:5432/postgres"
+/// task_id: 41524c07214502416f56b1258d97521b083483a0fbb082464e0741ace1403a2e
+/// role: Leader
+/// tolerable_clock_skew_seconds: 10
+/// hpke_config:
+///   id: 0
+///   kem_id: X25519HkdfSha256
+///   kdf_id: HkdfSha512
+///   aead_id: ChaCha20Poly1305
+///   public_key: a6f620ccc9d35164d92a4073727645f20329e1e1fe6bac7400c75a627c64783d
+/// "#;
+///
+/// let _decoded: AggregatorConfig = serde_yaml::from_str(yaml_config).unwrap();
+/// ```
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AggregatorConfig {
+    /// Address on which this server should listen for connections and serve its
+    /// API endpoints.
+    // TODO: Options for terminating TLS, unless that gets handled in a load
+    // balancer?
```

Comment:
FWIW, GCP's HTTP(S) load balancers do terminate TLS. Traffic from the GFE to the VM can be over TLS, but it seems this is mainly intended for compliance reasons and HTTP/2 compatibility. GFE does no certificate validation on backend TLS connections. They mention there is "automatic network-level encryption" transparently applied between GFE and the VM. https://cloud.google.com/load-balancing/docs/ssl-certificates/encryption-to-the-backends#encryption-to-backends